openedx/adg/lms/webinars/tests/test_admin.py

```diff
@@ -170,34 +172,72 @@ def test_send_update_emails_to_registrants_in_webinar_update(webinar_admin_insta
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('update', (True, False))
-def test_save_related_send_emails(request, webinar_admin_instance, webinar, mocker, update):
+@pytest.mark.parametrize('is_update, is_already_published, is_set_to_be_published', [
+    (False, False, False),
+    (False, False, True),
+    (True, False, True),
+    (True, True, True)
+])
+def test_save_related_send_emails(
+    request, webinar_admin_instance, mocker, is_update, is_already_published, is_set_to_be_published
+):
     """
-    Test that upon creating/updating webinar, emails for invites/updated invites are sent.
+    Test that the webinar updation or creation emails are only sent when a webinar is published.
+    No webinar email is sent for a non-published webinar
     """
     mock_send_webinar_emails = mocker.patch('openedx.adg.lms.webinars.admin.send_webinar_emails')
     mock_form_class = mocker.patch('openedx.adg.lms.webinars.forms.WebinarForm')
-
     mock_get_webinar_invitees_emails = mocker.patch('openedx.adg.lms.webinars.admin.get_webinar_invitees_emails')
-    mock_get_webinar_invitees_emails.return_value = ['test@email.com']
+    mock_get_webinar_invitees_emails.return_value = []
 
     request.method = 'POST'
-    mock_form_class.instance = webinar
-
-    if update:
-        user = UserFactory()
-        WebinarRegistrationFactory(user=user, webinar=webinar)
-        WebinarAdmin.save_related(webinar_admin_instance, request, mock_form_class, [], update)
-        mock_send_webinar_emails.assert_has_calls([
-            call(MandrillClient.WEBINAR_UPDATED, webinar, [user.email]),
-            call(MandrillClient.WEBINAR_CREATED, webinar, ['test@email.com'])
-        ])
+
+    presenter = UserFactory()
+    old_webinar = WebinarFactory(is_published=is_already_published, presenter=presenter)
+    webinar_admin_instance.old_webinar = old_webinar
+
+    new_webinar = WebinarFactory(is_published=is_set_to_be_published, presenter=presenter)
+    mock_form_class.instance = new_webinar
+
+    if is_update and is_already_published:
+        mock_get_webinar_invitees_emails.return_value = ['test@email.com']
+
+    WebinarAdmin.save_related(webinar_admin_instance, request, mock_form_class, [], is_update)
+
+    if is_update:
+        if is_set_to_be_published:
+            if is_already_published:
+                mock_send_webinar_emails.assert_has_calls([
+                    call(MandrillClient.WEBINAR_CREATED, new_webinar, ['test@email.com'])
+                ])
+            else:
+                mock_send_webinar_emails.assert_has_calls([
+                    call(MandrillClient.WEBINAR_CREATED, new_webinar, [presenter.email])
+                ])
+        else:
+            mock_send_webinar_emails.assert_not_called()
```

Comment:
The test will never execute this line with the current set of parameters. For this line to be executed, `is_update` has to be set to true, while `is_set_to_be_published` has to be set to False. There is no such parameter combination above this function.

Reply:
Good catch